src/platform/nrfconnect/wifi/ConnectivityManagerImplWiFi.cpp

```diff
@@ -39,124 +39,200 @@ using namespace ::chip::TLV;
+                        CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+
+    if (aMode != mStationMode)
+    {
+        mStationMode = aMode;
+        if (mStationMode != ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled)
+        {
+            bool doEnable{ ConnectivityManager::WiFiStationMode::kWiFiStationMode_Enabled == mStationMode };
+            if (doEnable)
+            {
+                OnStationConnected();
+            }
+            else
+            {
+                WiFiManager().Instance().DisconnectStation();
+                OnStationDisconnected();
+            }
+        }
+    }
+
     return CHIP_NO_ERROR;
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationEnabled(void)
 {
-    return false;
+    return (WiFiManager::StationStatus::DISABLED <= WiFiManager().Instance().GetStationStatus());
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationApplicationControlled(void)
 {
-    return false;
+    return (ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled == mStationMode);
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationConnected(void)
 {
-    return false;
+    return (WiFiManager::StationStatus::FULLY_PROVISIONED == WiFiManager().Instance().GetStationStatus());
 }
 
 System::Clock::Timeout ConnectivityManagerImplWiFi::_GetWiFiStationReconnectInterval(void)
 {
-    return System::Clock::kZero;
+    return mWiFiStationReconnectInterval;
 }
 
 CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationReconnectInterval(System::Clock::Timeout val)
 {
+    mWiFiStationReconnectInterval = val;
     return CHIP_NO_ERROR;
 }
 
 bool ConnectivityManagerImplWiFi::_IsWiFiStationProvisioned(void)
 {
-    return false;
+    return (WiFiManager::StationStatus::CONNECTED <= WiFiManager().Instance().GetStationStatus());
 }
 
-void ConnectivityManagerImplWiFi::_ClearWiFiStationProvision(void) {}
-
-ConnectivityManager::WiFiAPMode ConnectivityManagerImplWiFi::_GetWiFiAPMode(void)
+void ConnectivityManagerImplWiFi::_ClearWiFiStationProvision(void)
 {
-    return ConnectivityManager::WiFiAPMode::kWiFiAPMode_Disabled;
+    if (_IsWiFiStationProvisioned())
+    {
+        if (CHIP_NO_ERROR != WiFiManager().Instance().ClearStationProvisioningData())
+        {
+            ChipLogError(DeviceLayer, "Cannot clear WiFi station provisioning data");
+        }
+    }
 }
 
-CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiAPMode(ConnectivityManager::WiFiAPMode val)
+bool ConnectivityManagerImplWiFi::_CanStartWiFiScan()
 {
-    return CHIP_NO_ERROR;
+    return (WiFiManager::StationStatus::CONNECTING < WiFiManager().Instance().GetStationStatus());
 }
 
-bool ConnectivityManagerImplWiFi::_IsWiFiAPActive(void)
+void ConnectivityManagerImplWiFi::_OnWiFiStationProvisionChange()
 {
-    return false;
+    if (_IsWiFiStationProvisioned())
+    {
+        // probably the network provisioning data has been cleared so drop the connection
```

Comment:
It's probably unused, anyway, but I don't understand why we think it means disconnection. Perhaps, we can leave this method as unimplemented and consider removing it from Matter in the future.

Reply:
I am not sure how to interpret "provisioned" state. In WiFi world there are following steps involved when attaching the station: authenticating (checking if the device is 802.11 compliant),  associating (connecting to the network) and 4 way handshake for unicast secure communication (and group handshake for multicasting). Anyway this logic sucks and I am not sure how it is supposed to work (though that we should disconnect when the provisioning data is cleared), so maybe I will indeed just do nothing in _OnWiFiStationProvisionChange.

Reply:
Ok, let's now just make it work, ale later think about cleaning up the API. Given that our underlying API is not finalized yet, it probably doesn't make sense to spend too much time discussing it now. IMHO, "provisioned" in the case of Matter means that SSID&pass have been provided.